<launch>

 <!--base_control-->
   <node pkg="rosserial_python"
        type="serial_node.py" name="serial_node" >
    <param name="port" value="/dev/ttyACM0" />
    <param name="buad" value="115200" />
  </node>
 <!-- odom publisher and odom_to_base_linkã€€broadcaster-->
   <node pkg="smartcar"
        type="odompublisher" name="odom_publisher" >
  </node>

<arg name="camera" default="camera" />
	 <node pkg="mycamera"
        type="camera_splite" name="image_publisher" >
     </node>


    <node pkg="stereo_image_proc" type="stereo_image_proc" name="stereo_image_proc" ns="camera">
        <param name="stereo_algorithm" value="1"/>
        <param name="uniqueness_ratio" value="11.0"/>
        <param name="correlation_window_size" value="5.0"/>
        <param name="prefilter_size" value="5.0"/>
        <param name="prefilter_cap" value="63.0"/>
        <param name="min_disparity" value="0.0"/>
        <param name="disparity_range" value="64.0"/>
        <param name="speckle_size" value="98.0"/>
        <param name="texture_threshold" value="10.0"/>
     </node>

         <!-- run pointcloud_to_laserscan node -->
    <node pkg="pointcloud_to_laserscan" type="pointcloud_to_laserscan_node" name="pointcloud_to_laserscan">

        <remap from="cloud_in" to="$(arg camera)/points2"/>
        <remap from="scan" to="$(arg camera)/scan"/>
        <rosparam>
            target_frame: laser_link # Leave disabled to output scan in pointcloud frame
            transform_tolerance: 0.01
            min_height: 0.0
            max_height: 1.0

            angle_min: -1.5708 # -M_PI/2
            angle_max: 1.5708 # M_PI/2
            angle_increment: 0.0087 # M_PI/36045
            scan_time: 0.3333
            range_min: 0.45
            range_max: 4.0
            use_inf: false 

            # Concurrency level, affects number of pointclouds queued for processing and number of threads used
            # 0 : Detect number of cores
            # 1 : Single threaded
            # 2->inf : Parallelism level
            concurrency_level: 1
        </rosparam>

    </node>

 <!-- gmapping node -->

  <node pkg="gmapping" type="slam_gmapping" name="slam_gmapping">
    <remap from="scan" to="$(arg camera)/scan"/>
    <param name="delta" value="0.1"/>
    <param name="maxUrange" value="4.99"/>
       <param name="xmin" value="-5.0"/>
       <param name="ymin" value="-5.0"/>
    <param name="xmax" value="5.0"/>
    <param name="ymax" value="5.0"/>
    <param name="particles" value="60"/>
    <param name="srr" value="0"/>
    <param name="srt" value="0"/>
    <param name="str" value="0.05"/>
    <param name="stt" value="0.05"/>
    <param name="minimumScore" value="200"/>
    <param name="map_update_interval" value="1"/>
    <param name="lsigma" value="0.05"/>
  </node>  
 <node pkg="tf" type="static_transform_publisher" name="camera_link_to_laser_link" args="0.00 0 0.0 0 -1.57 1.57 camera_link laser_link 100"/>
 <node pkg="tf" type="static_transform_publisher" name="base_link_to_camera_link" args="0.05 0 0.3 -1.57 0 -1.57 base_link camera_link 100"/>
 <node pkg="image_view" type="stereo_view" name="stereo_view" >
    <remap from="stereo" to="$(arg camera)"/>
    <remap from="image" to="image_rect"/>
 </node> 
 <node pkg="rviz" type="rviz" name="rviz" >
 </node> 
</launch>
